src/e2e_test/e2e_test.rs

```diff
@@ -0,0 +1,291 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use astarte_sdk::types::AstarteType;
+use serde::{Deserialize, Serialize};
+use std::future::Future;
+use std::panic;
+
+use edgehog_device_runtime::e2e_test::{get_hardware_info, get_os_info, get_runtime_info};
+use edgehog_device_runtime::{DeviceManager, DeviceManagerOptions};
+
+#[derive(Serialize, Deserialize)]
+struct AstartePayload<T> {
+    data: T,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), edgehog_device_runtime::error::DeviceManagerError> {
+    let orig_hook = panic::take_hook();
+    panic::set_hook(Box::new(move |panic_info| {
+        println!("Test failed");
+        orig_hook(panic_info);
+        std::process::exit(1);
+    }));
+
+    //Waiting for Astarte Cluster to be ready...
+    tokio::time::sleep(std::time::Duration::from_millis(1000)).await;
+
+    let astarte_api_url = std::env::var("E2E_ASTARTE_API_URL").unwrap();
+    let realm = std::env::var("E2E_REALM_NAME").unwrap();
+    let device_id = std::env::var("E2E_DEVICE_ID").unwrap();
+    let credentials_secret = std::env::var("E2E_CREDENTIALS_SECRET").unwrap();
+    let pairing_url = astarte_api_url.to_owned() + "/pairing";
+    let e2e_token: &str = &std::env::var("E2E_TOKEN").unwrap();
+
+    let device_options = DeviceManagerOptions {
+        realm: realm.to_owned(),
+        device_id: Some(device_id.to_owned()),
+        credentials_secret: Some(credentials_secret),
+        pairing_url: pairing_url.to_string(),
+        pairing_token: None,
+        interfaces_directory: "./edgehog/astarte-interfaces".to_string(),
+        store_directory: "".to_string(),
+        download_directory: "".to_string(),
+    };
+
+    let mut dm = DeviceManager::new(device_options).await?;
+
+    dm.init().await?;
+
+    tokio::task::spawn(async move {
+        dm.run().await;
+    });
+
+    tokio::time::sleep(std::time::Duration::from_millis(2000)).await;
```

Comment:
Is this the waiting time for the `dm.run()` task? Maybe add a comment to clarify